File: airbyte-commons/src/test/java/io/airbyte/commons/text/NamesTest.java
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.commons.text;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

import org.junit.jupiter.api.Test;

class NamesTest {

  @Test
  void testToAlphanumericAndUnderscore() {
    assertEquals("users", Names.toAlphanumericAndUnderscore("users"));
    assertEquals("users123", Names.toAlphanumericAndUnderscore("users123"));
    assertEquals("UsErS", Names.toAlphanumericAndUnderscore("UsErS"));
    assertEquals("users_USE_special_____", Names.toAlphanumericAndUnderscore("users USE special !@#$"));
  }

  @Test
  void testSimpleQuote() {
    assertEquals("'abc'", Names.singleQuote("abc"));
    assertEquals("'abc'", Names.singleQuote("'abc'"));
    assertThrows(IllegalStateException.class, () -> Names.singleQuote("'abc"));
    assertThrows(IllegalStateException.class, () -> Names.singleQuote("abc'"));
  }

}

File: airbyte-commons/src/test/kotlin/io/airbyte/commons/text/NamesTest.kt
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.commons.text

import io.airbyte.commons.text.Names.singleQuote
import io.airbyte.commons.text.Names.toAlphanumericAndUnderscore
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.function.Executable

internal class NamesTest {
  @Test
  fun testToAlphanumericAndUnderscore() {
    Assertions.assertEquals("users", toAlphanumericAndUnderscore("users"))
    Assertions.assertEquals("users123", toAlphanumericAndUnderscore("users123"))
    Assertions.assertEquals("UsErS", toAlphanumericAndUnderscore("UsErS"))
    Assertions.assertEquals("users_USE_special_____", toAlphanumericAndUnderscore("users USE special !@#$"))
  }

  @Test
  fun testSimpleQuote() {
    Assertions.assertEquals("'abc'", singleQuote("abc"))
    Assertions.assertEquals("'abc'", singleQuote("'abc'"))
    Assertions.assertThrows<IllegalStateException?>(IllegalStateException::class.java, Executable { singleQuote("'abc") })
    Assertions.assertThrows<IllegalStateException?>(IllegalStateException::class.java, Executable { singleQuote("abc'") })
  }
}
